addons/mail/models/ir_model.py

```diff
@@ -100,3 +100,31 @@ def _instanciate(self, model_data):
             parents = [parents] if isinstance(parents, str) else parents
             model_class._inherit = parents + ['mail.thread.blacklist']
         return model_class
+
+    def _get_model_definitions(self, model_names_to_fetch):
+        response = {}
+        for model_name in model_names_to_fetch:
+            # get model fields, relational fields are kept only if the
+            # corresponding model is in model_names_to_fetch
+            model_fields = {
+                fname: field
+                for fname, field in self.env[model_name].fields_get(
+                    attributes=['name', 'type', 'relation', 'required', 'readonly', 'selection', 'string']).items()
+                if field.get('name') and (not field.get('relation') or field['relation'] in model_names_to_fetch)
```

Comment:
Some fields don't have a `name`?

Reply:
Yes, not many but some fields don't (`in_group_%`, `sel_group_%_%`).

Reply:
I'm not sure to understand why, and also why are we excluding them, surely they can be referenced somehow (using `fname` key instead of `name` value?)